### PR TITLE
Hubert / Fixing tick stream scroll and crash

### PR DESCRIPTION
**Pull request by hubert-deriv**

### CODE REVIEW

File: src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx [hunk 1]
```diff
@@ -31,6 +33,28 @@ export const PlaygroundComponent = () => {
...
   const location = useLocation();
   const history = createBrowserHistory();
+  
+  // If the user switches to a different tab, it will trigger the visibility state.
+  const documentVisibility = () => {
+    // If the visibility state is hidden, we will close the API.
+    if (document.visibilityState === "hidden") {
+      current_api.connection.close();
+      ticksSubject.complete();
+      document.removeEventListener("visibilitychange", documentVisibility, false);
```
Comment (yauheni-kryzhyk-deriv) [inline src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:43]: capture is false by default, so in our case we can remove third parameter in removeEventListener

File: src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx [hunk 2]
```diff
@@ -23,6 +23,8 @@ export const PlaygroundComponent = () => {
   const request_input = useRef(null);
   const [request_info, setRequestInfo] = useState({});
   const [response_info, setResponseInfo] = useState({});
+  const [visibility_listener, setVisibilityListener] = useState(true);
+  const [scroll_direction, setScrollDirection] = useState("down");
```
Comment (yauheni-kryzhyk-deriv) [inline src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:27]: just a nitpick: do we use in this project double quotes?

**Review by sanjam-deriv — APPROVED** [review]

File: src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx [hunk 3]
```diff
@@ -31,6 +33,28 @@ export const PlaygroundComponent = () => {
...
+    setCurrentAPI(api);
+  }
+
+  // Here we initialize the visibility listener.
+  useEffect(() => {
+    if (visibility_listener) {
+      setVisibilityListener(false);
+      document.addEventListener("visibilitychange", documentVisibility, false);
+    }
+  }, [visibility_listener])
```
Comment (yauheni-kryzhyk-deriv) [inline src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:57]: Should we also delete event listener after unmounting component?

Comment (hubert-deriv) [reply src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:57]: This is a very good point. Yes indeed, we should. I will to add something in regards to that. 👍

File: src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx (hunk 2 above)
Comment (hubert-deriv) [reply src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:27]: Haha, I understand. Well, in this case I prefer double quotes as this seems to be the standard within this project and thus would be more consistent.

File: src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx (hunk 1 above)
Comment (hubert-deriv) [reply src/components/playground/PlaygroundComponent/PlaygroundComponent.jsx:43]: Good point. Will remove it.

**Pull request merged**